@stellar/typescript-wallet-sdk/src/walletSdk/Auth/AuthHeaderSigner.ts

```diff
@@ -0,0 +1,161 @@
+import { AxiosInstance } from "axios";
+import { StrKey } from "@stellar/stellar-sdk";
+import nacl from "tweetnacl";
+import naclUtil from "tweetnacl-util";
+import base64url from "base64url";
+
+import { SigningKeypair } from "../Horizon/Account";
+import { DefaultClient } from "../";
+import { AuthHeaderClaims, AuthHeaderCreateTokenParams } from "../Types";
+import {
+  AuthHeaderSigningKeypairRequiredError,
+  AuthHeaderClientDomainRequiredError,
+} from "../Exceptions";
+
+export interface AuthHeaderSigner {
+  createToken({
+    claims,
+    clientDomain,
+    issuer,
+  }: AuthHeaderCreateTokenParams): Promise<string>;
+}
+
+/**
+ * Signer for signing JWT for GET /Auth with a custodial private key
+ *
+ * @class
+ */
+export class DefaultAuthHeaderSigner implements AuthHeaderSigner {
+  expiration: number;
+
+  constructor(expiration: number = 900) {
+    this.expiration = expiration;
+  }
+
+  /**
+   * Create a signed JWT for the auth header
+   * @constructor
+   * @param {AuthHeaderCreateTokenParams} params - The create token parameters
+   * @param {AuthHeaderClaims} params.claims - the data to be signed in the JWT
+   * @param {string} [params.clientDomain] - the client domain hosting SEP-1 toml
+   * @param {AccountKeypair} [params.issuer] - the account signing the JWT
+   * @returns {Promise<string>} The signed JWT
+   */
+  // eslint-disable-next-line @typescript-eslint/require-await
+  async createToken({
+    claims,
+    // eslint-disable-next-line @typescript-eslint/no-unused-vars
+    clientDomain,
+    issuer,
+  }: AuthHeaderCreateTokenParams): Promise<string> {
+    if (!(issuer instanceof SigningKeypair)) {
+      throw new AuthHeaderSigningKeypairRequiredError();
+    }
+
```

Comment:
would love extra 👀 on this JWT creation code using nacl, to make sure it looks right

note: web api crypto [doesn't support ed25519](https://developer.mozilla.org/en-US/docs/Web/API/SubtleCrypto/importKey#browser_compatibility) for most browsers, which is why I'm using nacl, which we're also using in the keymanager code